presto-main/src/main/java/com/facebook/presto/resourcemanager/DistributedResourceGroupInfoResource.java

```diff
@@ -0,0 +1,155 @@
+            asyncResponse.resume(Response.status(NOT_FOUND).build());
+        }
+        try {
+            ImmutableList.Builder<ListenableFuture<ResourceGroupInfo>> queryStateInfoFutureBuilder = ImmutableList.builder();
+            for (InternalNode coordinator : internalNodeManager.getCoordinators()) {
+                queryStateInfoFutureBuilder.add(getResourceGroupInfoFromCoordinator(xForwardedProto, uriInfo, coordinator));
+            }
+            List<ListenableFuture<ResourceGroupInfo>> queryStateInfoFutureList = queryStateInfoFutureBuilder.build();
+            Futures.whenAllComplete(queryStateInfoFutureList).call(() -> {
+                try {
+                    ResourceGroupInfo aggregatedResourceGroupInfo = aggregateResourceGroupInfo(queryStateInfoFutureList);
+                    if (aggregatedResourceGroupInfo == null) {
+                        return asyncResponse.resume(Response.status(NOT_FOUND).build());
+                    }
+                    return asyncResponse.resume(Response.ok(aggregatedResourceGroupInfo).build());
+                }
+                catch (Exception ex) {
+                    log.error(ex, "Error in getting resource group info from one of the coordinators");
+                    return asyncResponse.resume(Response.serverError().entity(ex.getMessage()).build());
+                }
+            }, executor);
+        }
+        catch (IOException ex) {
+            log.error(ex, "Error in getting resource group info");
+            asyncResponse.resume(Response.serverError().entity(ex.getMessage()).build());
+        }
+    }
+
+    private ResourceGroupInfo aggregateResourceGroupInfo(List<ListenableFuture<ResourceGroupInfo>> queryStateInfoFutureList)
+            throws InterruptedException, ExecutionException
+
+    {
+        Iterator<ListenableFuture<ResourceGroupInfo>> iterator = queryStateInfoFutureList.iterator();
+        AggregatedResourceGroupInfoBuilder builder = new AggregatedResourceGroupInfoBuilder();
+        while (iterator.hasNext()) {
+            try {
+                builder.add(iterator.next().get());
+            }
+            catch (ExecutionException e) {
+                Throwable exceptionCause = e.getCause();
+                //airlift JsonResponseHandler throws UnexpectedResponseException for cases where http status code != 2xx
+                if (!(exceptionCause instanceof UnexpectedResponseException) ||
+                        ((UnexpectedResponseException) exceptionCause).getStatusCode() != NOT_FOUND.getStatusCode()) {
+                    throw e;
+                }
+            }
+        }
+        return builder.build();
+    }
+
+    private ListenableFuture<ResourceGroupInfo> getResourceGroupInfoFromCoordinator(String xForwardedProto, UriInfo uriInfo,
+            InternalNode coordinatorNode)
+            throws IOException
+    {
+        String scheme = isNullOrEmpty(xForwardedProto) ? uriInfo.getRequestUri().getScheme() : xForwardedProto;
```

Comment:
i don't see us supporting xForwardedProto anywhere else. Do we know if it's been used or not?

Reply:
I think its not currently used in the resource group flow